plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1645,8 +1647,9 @@ private void finishOptimize(ConnectorSession session, DeltaLakeTableExecuteHandl
 
             long writeTimestamp = Instant.now().toEpochMilli();
 
-            for (Path scannedPath : scannedPaths) {
-                String relativePath = new Path(tableLocation).toUri().relativize(scannedPath.toUri()).toString();
+            URI tableLocationUri = Locations.toUri(tableLocation);
+            for (String scannedPath : scannedPaths) {
+                String relativePath = tableLocationUri.relativize(Locations.toUri(scannedPath)).toString();
```

Comment:
We could add a `relativePath(root, path)` function that verifies that `root` is a prefix of `path`.